File: src/main/java/com/cag/cagbackendapi/entities/EthnicityMemberEntity.java
package com.cag.cagbackendapi.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import java.util.UUID;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "ethnicity_member")
public class EthnicityMemberEntity {
    @Id
    @Type(type = "pg-uuid")
    private UUID ethnicity_member_id;

    @ManyToOne
    @JoinColumn(foreignKey = @ForeignKey(name = "fk_profile"))
    private ProfileEntity profileEntity;

    @ManyToOne
    @JoinColumn(foreignKey = @ForeignKey(name = "fk_ethnicity"))
    private EthnicityEntity ethnicityEntity;
}

File: src/main/java/com/cag/cagbackendapi/entities/EthnicityMemberEntity.kt
package com.cag.cagbackendapi.entities

import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.Type
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "ethnicity_member")
data class EthnicityMemberEntity (
    @Id
    @Type(type = "pg-uuid")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator"
    )
    var ethnicity_member_id: UUID?,

    @ManyToOne
    @JoinColumn(name = "profile_id", referencedColumnName = "profile_id")
    var profileEntity: ProfileEntity?,

    @ManyToOne
    @JoinColumn(name = "ethnicity_id", referencedColumnName = "ethnicity_id")
    var ethnicityEntity: EthnicityEntity?
)
